docs/debugging-testing/testing/unit_testing.md

```diff
@@ -61,19 +61,21 @@ In a terminal window:
 
 ## Test code structure
 
-Find unit tests in the Mbed OS repository under the `UNITTESTS` folder of each library. We recommend unit test files use an identical directory path as the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file you're testing is `some/example/path/ClassName.cpp`, then all the test files are in the `UNITTESTS/some/example/path/ClassName` directory. Each test suite needs to have its own `unittest.cmake` file for test configuration.
+Find unit tests in the Mbed OS repository under the `UNITTESTS` folder of each library. We recommend unit test files use an identical directory path as the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file you're testing is `some/example/path/ClassName.cpp`, then all the test files are in the `UNITTESTS/some/example/path/ClassName` directory. Each test suite needs to have its own `CMakeLists.txt` file for test CMake configuration.
 
-All the class stubs should be located in the Mbed OS root directory `UNITTESTS/stubs`. Multiple test suites can use a single stub class, which should follow the naming convention `ClassName_stub.cpp` for the source file and `ClassName_stub.h` for the header file. Use the actual header files for the unit tests, and don't stub headers if possible. The stubbed headers reside in the `UNITTESTS/target_h` directory.
+All the stub sources are grouped into stub libraries `mbed-stubs-rtos`, `mbed-stubs-drivers`, `mbed-stubs-events`, `mbed-stubs-hal`, `mbed-stubs-platform`, `mbed-stubs-storage`, `mbed-stubs-connectivity` and all stub libraries are target linked under `mbed-stubs`. The library unit test suite CMake is expected to link with the required stub library or directly link with mbed-stubs depends on their test cases.
```

Comment:
Is the list exhaustive? Why not give just one example.
```suggestion
All the stub sources are built in stub CMake library targets (e.g `mbed-stubs-rtos`) and linked to the `mbed-stubs` CMake target. The CMake target of the library unit under test is expected to link with the required stub libraries or `mbed-stubs` depending on the test case. 
```

`...depending on the test case`, can you explain a little bit when to do one or the other?

Reply:
For watchdog test cases requires only mbed-stubs-platform, mbed-stubs-hal but connectivity test cases require multiple stub libraries so better to go with mbed-stubs

Reply:
Please add your explanation in the readme.